host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -199,6 +202,34 @@ static esp_err_t _uac_host_device_delete(uac_device_t *uac_device);
 static esp_err_t uac_cs_request_set(uac_device_t *uac_device, const uac_cs_request_t *req);
 static esp_err_t uac_cs_request_set_ep_frequency(uac_iface_t *iface, uint8_t ep_addr, uint32_t freq);
 
+// --------------------------- Utility Functions --------------------------------
+/**
+ * @brief Convert volume U16 type to UAC volume db, for details, please refer to UAC 1.0 spec P.77
+ * 0x7FFF: 127.9961db
+ * 0x0100: 1.0000db
+ * 0x0000: 0.0000db
+ * 0xFF00: -1.0000db (there is an typo in UAC 1.0 spec, it should be 0xFF00)
+ * 0xFE00: -2.0000db
+ * 0x8001: -127.9961db
+ * @param[in] volume      Volume in db
+ * @return uint16_t
+ */
+static float _volume_db_u16_2_f(uint16_t volume_db)
+{
+    return (volume_db & 0x8000 ? -128.0 : 0) + ((uint16_t)(volume_db & 0x7fff)) * 1.0 / 256;
```

Comment:
```suggestion
    return (volume_db & 0x8000 ? -128.0 : 0) + ((uint16_t)(volume_db & 0x7fff)) / 256.0f;
```
Can we do it like this? This way we will avoid multiplication by 1. 

Also, I am not sure if we have to explicitly set floating point numbers to float type, like this: `256.0f` or if the compiler would do it automatically.

Reply:
@peter-marcisovsky the 256 will be implicitly converted to 256.0f

 for your reference: https://en.cppreference.com/w/c/language/conversion

![image](https://github.com/user-attachments/assets/e40fbf8a-ce50-4aec-bd43-cbe9e6f93143)

Reply:
To avoid confusion, I've added an `f` suffix to all decimals

Reply:
Thx for the reference. Looks good.